src/internal/packager/helm/zarf.go

```diff
@@ -72,3 +77,85 @@ func (h *Helm) UpdateZarfGiteaValues() error {
 
 	return nil
 }
+
+// UpdateZarfAgentValues updates the Zarf git server deployment with the new state values
+func (h *Helm) UpdateZarfAgentValues() error {
+	spinner := message.NewProgressSpinner("Gathering information to update Zarf Agent TLS")
+	defer spinner.Stop()
+
+	err := h.createActionConfig(cluster.ZarfNamespaceName, spinner)
+	if err != nil {
+		return fmt.Errorf("unable to initialize the K8s client: %w", err)
+	}
+
+	// Get the current agent image from one of its pods.
+	pods := h.Cluster.WaitForPodsAndContainers(k8s.PodLookup{
+		Namespace: cluster.ZarfNamespaceName,
+		Selector:  "app=agent-hook",
+	}, nil)
+
+	var currentAgentImage transform.Image
+	if len(pods) > 0 && len(pods[0].Spec.Containers) > 0 {
+		currentAgentImage, err = transform.ParseImageRef(pods[0].Spec.Containers[0].Image)
+		if err != nil {
+			return fmt.Errorf("unable to parse current agent image reference: %w", err)
+		}
+	} else {
+		return fmt.Errorf("unable to get current agent pod")
+	}
+
+	// List the releases to find the current agent release name.
+	listClient := action.NewList(h.actionConfig)
+
+	releases, err := listClient.Run()
+	if err != nil {
+		return fmt.Errorf("unable to list helm releases: %w", err)
+	}
+
+	spinner.Success()
+
+	for _, lsRelease := range releases {
+		// Update the Zarf Agent release with the new values
+		if lsRelease.Chart.Name() == "raw-init-zarf-agent-zarf-agent" {
+			h.Chart = types.ZarfChart{
+				Namespace: "zarf",
+			}
+			h.ReleaseName = lsRelease.Name
+			h.Component = types.ZarfComponent{
+				Name: "zarf-agent",
+			}
+			h.Cfg.Pkg.Constants = []types.ZarfPackageConstant{
+				{
+					Name:  "AGENT_IMAGE",
+					Value: currentAgentImage.Path,
+				},
+				{
+					Name:  "AGENT_IMAGE_TAG",
+					Value: currentAgentImage.Tag,
+				},
+			}
+
+			err := h.UpdateReleaseValues(map[string]interface{}{})
+			if err != nil {
+				return fmt.Errorf("error updating the release values: %w", err)
+			}
+		}
+	}
+
+	spinner = message.NewProgressSpinner("Cleaning up Zarf Agent pods after update")
+	defer spinner.Stop()
+
+	// Force pods to be recreated to get the updated secret.
+	pods = h.Cluster.WaitForPodsAndContainers(k8s.PodLookup{
+		Namespace: cluster.ZarfNamespaceName,
+		Selector:  "app=agent-hook",
+	}, nil)
+
```

Comment:
Is there a reason we are making another call to `WaitForPodsAndContainers` here? At this stage we already have the pods in-memory and should be able to iterate over them and delete them without re-fetching.

```suggestion
```

Reply:
I did this because the update command takes some time and there is a _chance_ that one of the pods was recycled behind the scenes and we no longer hold a reference to it - we could look at doing this better though - will think more on it, maybe adding a delete by selector or something.

Reply:
Just updated to remove this in favor of a DeletePods function.